src/foraging_gui/RigJsonBuilder.py

```diff
@@ -0,0 +1,510 @@
+                firmware_version="FTDI version:",
+                computer_name="behavior_computer", 
+                is_clock_generator=False,
+                channels=[
+                    d.DAQChannel(channel_name="DO0", device_name="Solenoid Left", channel_type="Digital Output"),
+                    d.DAQChannel(channel_name="DO1", device_name="Solenoid Right", channel_type="Digital Output"),
+                    d.DAQChannel(channel_name="DI0", device_name=lick_spouts[0].name, channel_type="Digital Input"),
+                    d.DAQChannel(channel_name="DI1", device_name=lick_spouts[1].name, channel_type="Digital Input"),
+                    d.DAQChannel(channel_name="DI3", device_name="Photometry Clock", channel_type="Digital Input"),
+                ],
+            )
+        ]
+    else:
+        components['daqs']=[
+            d.HarpDevice(
+                name="Harp Behavior",
+                harp_device_type=d.HarpDeviceType.BEHAVIOR,
+                core_version="2.1",
+                firmware_version="FTDI version:",
+                computer_name="behavior_computer", 
+                is_clock_generator=False,
+                channels=[
+                    d.DAQChannel(channel_name="DO0", device_name="Solenoid Left", channel_type="Digital Output"),
+                    d.DAQChannel(channel_name="DO1", device_name="Solenoid Right", channel_type="Digital Output"),
+                    d.DAQChannel(channel_name="DI0", device_name=lick_spouts[0].name, channel_type="Digital Input"),
+                    d.DAQChannel(channel_name="DI1", device_name=lick_spouts[1].name, channel_type="Digital Input") 
+                ],
+            )
+        ]
+
+    # Optogenetics specific
+    ###########################################################################
+    if OPTO:
+        ##Optogenetics Specific   ## TODO Xinxin to fill in
+        print('need to implement') 
+        #components['light_sources'].append(
+        #    d.LightEmittingDiode(
+        #        name="LED for photostimulation",
+        #        manufacturer=d.Organization.PRIZMATIX,
+        #        model="xxx",
+        #        wavelength=470,
+        #    )
+        #    )
+        
+        #daqs.append(
+        #    d.DAQDevice(
+        #        name="NIDAQ for opto",
+        #        device_type="DAQ Device",
+        #        data_interface="USB2.0",
+        #        manufacturer=d.Organization.NATIONAL_INSTRUMENTS,
+        #        computer_name="behavior_computer",
+        #        channels=[
+        #        ],
+        #    )
+        #    )
+
+    # Generate Rig Schema
+    ###########################################################################
+    # Assemble rig schema
+    rig = r.Rig(
+        rig_id="{}_{}".format(settings['rig_name'],datetime.now().strftime('%Y-%m-%d')), 
+        modification_date=date.today(),
+        **components 
+        )
+    logging.info('finished building rig json')
+    return rig
+
```

Comment:
There are other devices (e.g. harp devices, USB daq for optogenetics and IR light) we need to add.

Reply:
(just copy pasting a post in Teams chat) I believe I have communicated with you that my template lacks optogenetics-related stuff and need to be added.

Reply:
I've added these additional devices